src/H5Tconv_bitfield.c

```diff
@@ -161,8 +163,7 @@ H5T__conv_b_b(const H5T_t *src, const H5T_t *dst, H5T_cdata_t *cdata, const H5T_
                 if (src->shared->u.atomic.prec > dst->shared->u.atomic.prec) {
                     /*overflow*/
                     if (conv_ctx->u.conv.cb_struct.func) { /*If user's exception handler is present, use it*/
-                        H5T__reverse_order(src_rev, s, src->shared->size,
-                                           src->shared->u.atomic.order); /*reverse order first*/
+                        H5T__reverse_order(src_rev, s, src); /*reverse order first*/
```

Comment:
Since byte-swapping a complex number value involves swapping the real and imaginary parts separately, `H5T__reverse_order` was updated to just take a pointer to the datatype involved so it can do the right thing.